platform/engineering/request-for-change/2019-09-30-cms-integration.md

```diff
@@ -0,0 +1,226 @@
+# RFC: CMS Integration
+
+- Date: _2019-09-30_
+- Related Issue:
+
+## Background
+_Explain the current state of the feature._
+
+### Glossary
+This RFC will use the following terms. These terms may or may not be
+used outside of this RFC.
+- Full Jenkins build
+  - All branches are build using this pipeline (including `master`)
+  - This attempts to pull all CMS content, but will fall back to the
+    cache if necessary
+  - This builds JavaScript and CSS assets
+    - For the build of the `master` branch, these assets are saved in
+      S3
+  - Runs the full suite of unit, e2e, and accessibility tests
+- Content-only build
+  - Used in a content-only deploy
+  - Attempts to pull the latest CMS content
+    - Will not fall back to the cache
+  - Uses the JavaScript and CSS assets from the latest full build
+  - Runs a limited suite of tests
+
+### Metalsmith
+Currently, `vets-website`'s build is managed by Metalsmith, a
+plugin-based static site generator. Metalsmith uses a number of custom
+plugins used to manipulate the static content in various ways. One
+such plugin fetches the content from Drupal if we pass the
+`--pull-drupal` flag to the build script or if the local cache isn't
+found, another checks for broken links, etc.
+
+This build script is run in a number of ways:
+- The Jenkins pipeline for a full build
+- The Jenkins pipeline for a content-only build
+- Locally, with `yarn build` or `yarn watch`
+
+Separate, but related, the preview server has a modified version of
```

Comment:
What's the preview server? (I _kind of_ know what it is, but maybe a screenshot, link to the server, or document link would help)

Reply:
This is a great question. The short answer is in the sentence following, but I realized yesterday that I'm actually not sure where content editors actually _go_ to see their changes. They're likely not running the preview server locally. :man_shrugging:

That said, it's mostly tangential to this RFC. Just thought it would be worth mentioning since it exists and is related.

I'll see if we have any documentation floating around for the preview server and link to it if we do. :+1: